### PR TITLE
Adjust logging in gateway

### PR DESCRIPTION
**Pull request by alextargov**
**Description**

Changes proposed in this pull request:

- Replace the default logger with the Director's one
- Add correlation ids to the logs
- Revise the log levels

**Related issue(s)**

### CODE REVIEW

**Review by NickyMateev — DISMISSED** [review]
LGTM 👍

File: components/gateway/cmd/main.go [hunk 1]
```diff
@@ -247,21 +245,23 @@ func fillJWTCredentials(cfg auditlog.OAuthConfig) clientcredentials.Config {
 	}
 }
 
-func initWorkers(workers chan bool, auditlogSvc proxy.AuditlogService, done chan bool, msgChannel chan proxy.AuditlogMessage, collector *metrics.AuditlogCollector) {
-	go func() {
+func initWorkers(ctx context.Context, workers chan bool, auditlogSvc proxy.AuditlogService, done chan bool, msgChannel chan proxy.AuditlogMessage, collector *metrics.AuditlogCollector) {
+	logger := log.C(ctx)
+
+	go func(log *logrus.Entry) {
```
Comment (DimitarPetrov) [inline components/gateway/cmd/main.go:251]: Why passing the logger to the goroutine is needed when it is available in the closure?

Comment (alextargov) [reply components/gateway/cmd/main.go:251]: Well spotted. It was not intended really.

**Review by DimitarPetrov — APPROVED** [review]

**Pull request merged**